src/Adapter/Attachment/AttachmentRepository.php

```diff
@@ -59,6 +83,51 @@ public function get(AttachmentId $attachmentId): Attachment
         return $attachment;
     }
 
+    /**
+     * @param ProductId $productId
+     *
+     * @return array<int, array<string, string|array<int, string>>>
+     */
+    public function getAllByProduct(ProductId $productId): array
```

Comment:
```suggestion
    public function getProductAttachments(ProductId $productId): array
```

or maybe more simple

```suggestion
    public function getByProduct(ProductId $productId): array
```

(the `getAll` feels weird since it's a filter by product)